Sources/CombineBonjour/NetworkServiceBrowser.swift

```diff
@@ -0,0 +1,197 @@
+        // Only demand starts the side-effect, so we have to be very lazy and postpone the side-effects as much as possible
+        private var started: Bool = false
+
+        init(subscriber: SubscriberType, serviceType: String, domain: String?) {
+            let params: NWParameters = .init() // udp
+            params.prohibitExpensivePaths = true // do not browse WAN
+            self.browser = NWBrowser(for: NWBrowser.Descriptor.bonjourWithTXTRecord(type: serviceType,
+                                                                                    domain: domain),
+                                     using: params)
+            self.buffer = DemandBuffer(subscriber: subscriber)
+            self.domain = domain
+            self.serviceType = serviceType
+
+            browser.stateUpdateHandler = .some({ [weak self] state in
+                switch state {
+                case .setup:
+                    break
+                case .ready:
+                    break
+                case let .failed(error):
+                    _ = self?.buffer?.complete(completion: .failure(NetworkServiceBrowser.NetworkServiceBrowserError.didNotSearch(error: error)))
+                case .cancelled:
+                    break
+                case let .waiting(error):
+                    if case let .dns(dnsServiceErrorType) = error, dnsServiceErrorType == kDNSServiceErr_PolicyDenied {
+                        _ = self?.buffer?.complete(completion: .failure(NetworkServiceBrowser.NetworkServiceBrowserError.bonjourPermissionDenied))
+                    } else {
+                        _ = self?.buffer?.complete(completion: .failure(NetworkServiceBrowser.NetworkServiceBrowserError.didNotSearch(error: error)))
+                    }
+                @unknown default:
+                    break
+                }
+
+            })
+            browser.browseResultsChangedHandler = .some({ [weak self] (_ , changes: Set<NWBrowser.Result.Change>) in
+                changes.forEach { change in
+                    switch change {
+                    case .identical:
+                        break
+                    case let .added(result):
+                        var txtRecord: NWTXTRecord? = nil
+                        if case let .bonjour(txt) = result.metadata {
+                            txtRecord = txt
+                        }
+                        _ = self?.buffer?.buffer(value: .init(type: .didFind(name: serviceType, domain: domain, result: result, txt: txtRecord)))
+                    case let .removed(result):
+                        _ = self?.buffer?.buffer(value: .init(type: .didRemove(name: serviceType, domain: domain, result: result)))
+                    case .changed(old: let old, new: let new, flags: let flags):
+                        _ = self?.buffer?.buffer(value: .init(type: .didUpdate(name: serviceType, domain: domain, old: old, new: new, flags: flags)))
```

Comment:
On updated don't we want to extract the possible TXT Records the same way we do for on added? I guess when we first find the service, we haven't interrogated it yet about its metadata (and IP address), so when using NetServiceDelegate we have to explicitly request the TXT Records update which will come later as an update to the NetService. I don't know if this is also the case here.